### PR TITLE
[ci] add pre-commit checks to CI

### LINKED ISSUE #403: [CI] run pre-commit checks on CI
After #393 is completed, add support for running the pre-commit checks on CI.

### PR DESCRIPTION
**Pull request by pilkicTT**
Run pre-commit checks as part of CI.

Closes #403

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  pre-commit:
+
+    timeout-minutes: 10
+    runs-on: macos-latest
```
Comment (nvukobratTT) [inline .github/workflows/pre-commit.yml:11]: Do we need to run on MacOS specifically? 

I assume we'll have to deprecate it at some point, so maybe it'll be better to use Linux OS flavor

**Comment by pilkicTT** [comment]
Pre-commit workflow is failing for now; it should pass after 
- #405 

is merged...

File: .github/workflows/pre-commit.yml (hunk 1 above)
Comment (pilkicTT) [reply .github/workflows/pre-commit.yml:11]: Good catch!

**Review by vmilosevic — APPROVED** [review]
Looks good

**Comment by pilkicTT** [comment]
Merging now to start enforcing this ASAP.

**Pull request merged**